cudax/include/cuda/experimental/__async/continue_on.cuh

```diff
@@ -0,0 +1,286 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDAX_ASYNC_DETAIL_CONTINUE_ON_H
+#define __CUDAX_ASYNC_DETAIL_CONTINUE_ON_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/experimental/__async/completion_signatures.cuh>
+#include <cuda/experimental/__async/cpos.cuh>
+#include <cuda/experimental/__async/exception.cuh>
+#include <cuda/experimental/__async/meta.cuh>
+#include <cuda/experimental/__async/queries.cuh>
+#include <cuda/experimental/__async/tuple.cuh>
+#include <cuda/experimental/__async/utility.cuh>
+#include <cuda/experimental/__async/variant.cuh>
+
+#include <cuda/experimental/__async/prologue.cuh>
+
+namespace cuda::experimental::__async
+{
+struct continue_on_t
+{
+#ifndef __CUDACC__
+
+private:
+#endif
+  template <class... _As>
+  using __set_value_tuple_t = __tuple<set_value_t, __decay_t<_As>...>;
+
+  template <class _Error>
+  using __set_error_tuple_t = __tuple<set_error_t, __decay_t<_Error>>;
+
+  using __set_stopped_tuple_t = __tuple<set_stopped_t>;
+
+  using __complete_fn = void (*)(void*) noexcept;
+
+  template <class... _Ts>
+  using __set_value_completion =
+    __mif<__nothrow_decay_copyable<_Ts...>,
```

Comment:
What is the difference between this an `cuda::std::_If`

Reply:
no difference. pointless duplication.

Reply:
actually, i kinda take that back. `__mif` does double-duty as `_If` and as `enable_if_t`. `__mif<C>` and `__mif<C, T>` behave like `enable_if_t`, while `__mif<C,T,U>` behaves like `_If<C,T,U>`.

but mostly its pointless duplication. i imagine there is a lot of overlap with CCCL utilities, and fully CCCL-ifying this code will take a long time, and probably won't be done in this PR.